### PR TITLE
[W5.11][F09-A3]Wu Di

### PR DESCRIPTION
**Pull request by WoodySIN**
add two feature: 1) sort command; 2) find command case insensitive

### CODE REVIEW

File: src/seedu/addressbook/commands/FindCommand.java [hunk 1]
```diff
@@ -41,7 +41,7 @@ public CommandResult execute() {
     }
 
     /**
-     * Retrieves all persons in the address book whose names contain some of the specified keywords.
+     * Retrieves all persons in the address book whose names contain some of the specified keywords regardless of case.
```
Comment (RyanAngJY) [inline src/seedu/addressbook/commands/FindCommand.java:44]: Can use "case-insensitive"

File: src/seedu/addressbook/commands/Command.java [hunk 2]
```diff
@@ -36,6 +36,16 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
         return String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, personsDisplayed.size());
     }
 
+    /**
+     * Constructs a feedback message to summarise an operation that displayed a listing of persons.
+     *
+     * @param personsDisplayed used to generate summary
+     * @return summary message for persons displayed
+     */
+    public static String getMessageForPersonSortedListShownSummary(List<? extends ReadOnlyPerson> personsDisplayed) {
```
Comment (RyanAngJY) [inline src/seedu/addressbook/commands/Command.java:45]: Method name is slightly confusing

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 3]
```diff
@@ -72,6 +72,13 @@ public UniquePersonList(UniquePersonList source) {
         internalList.addAll(source.internalList);
     }
 
+    /**
+     * Sorts the list by natural ordering of names
+     */
+    public void sort() {
```
Comment (RyanAngJY) [inline src/seedu/addressbook/data/person/UniquePersonList.java:78]: Method name can be slightly more descriptive

File: src/seedu/addressbook/commands/SortCommand.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package seedu.addressbook.commands;
```
Comment (RonakLakhotia) [inline src/seedu/addressbook/commands/SortCommand.java:1]: Missing java docs.

File: src/seedu/addressbook/commands/SortCommand.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+import seedu.addressbook.data.person.*;
```
Comment (RonakLakhotia) [inline src/seedu/addressbook/commands/SortCommand.java:3]: Avoid using * to import all classes from the package. Import the classes individually

File: src/seedu/addressbook/data/person/Name.java [hunk 6]
```diff
@@ -37,10 +37,10 @@ public static boolean isValidName(String test) {
     }
 
     /**
-     * Retrieves a listing of every word in the name, in order.
+     * Retrieves a listing of every word in the name, in order and makes it lower case.
      */
     public List<String> getWordsInName() {
-        return Arrays.asList(fullName.split("\\s+"));
+        return Arrays.asList(fullName.toLowerCase().split("\\s+"));
```
Comment (RonakLakhotia) [inline src/seedu/addressbook/data/person/Name.java:43]: use \\s+ in a separate variable, avoid using magic numbers

File: src/seedu/addressbook/data/person/Person.java [hunk 7]
```diff
@@ -8,7 +8,7 @@
  * Represents a Person in the address book.
  * Guarantees: details are present and not null, field values are validated.
  */
-public class Person implements ReadOnlyPerson {
+public class Person implements ReadOnlyPerson, Comparable<Person>{
```
Comment (RonakLakhotia) [inline src/seedu/addressbook/data/person/Person.java:11]: Leave a space before {

File: src/seedu/addressbook/data/person/Person.java [hunk 8]
```diff
@@ -73,6 +73,10 @@ public boolean equals(Object other) {
                 && this.hasSameData((ReadOnlyPerson) other));
     }
 
+    public int compareTo(Person other) {
```
Comment (RonakLakhotia) [inline src/seedu/addressbook/data/person/Person.java:76]: use a better variable name than other

**Review by RonakLakhotia — COMMENTED** [review]
Good attempt. Keep in mind the coding standard violations

**Pull request closed without merging**